tests/transfer_item.test.js

```diff
@@ -0,0 +1,134 @@
+const { TezosToolkit, MichelsonMap } = require("@taquito/taquito");
+const { InMemorySigner } = require("@taquito/signer");
+
+const inventoryContract = require("../build/contracts/inventory.json");
+const warehouseContract = require("../build/contracts/warehouse.json");
+
+const {
+    getInventoryItemAt,
+    hasInventoryItemAt,
+    originateContract
+} = require("./utils");
+
+describe("Given Warehouse and Inventory are deployed", () => {
+    let warehouseInstance;
+    let oldInventoryInstance;
+    let newInventoryInstance;
+    let tezos;
+
+    beforeAll(async () => {
+        tezos = new TezosToolkit("http://localhost:20000");
+
+        tezos.setProvider({
+            signer: new InMemorySigner(
+                "edsk3QoqBuvdamxouPhin7swCvkQNgq4jP5KZPbwWNnwdZpSpJiEbq"
+            )
+        });
+
+        warehouseInstance = await originateContract(tezos, warehouseContract, {
+            owner: "tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb",
+            version: "1",
+            warehouse: MichelsonMap.fromLiteral({})
+        });
+
+        oldInventoryInstance = await originateContract(
+            tezos,
+            inventoryContract,
+            MichelsonMap.fromLiteral({})
+        );
+
+        newInventoryInstance = await originateContract(
+            tezos,
+            inventoryContract,
+            MichelsonMap.fromLiteral({})
+        );
+    });
+
+    describe("When I add a new item with a quantity of 1", () => {
+        beforeAll(async () => {
+            const operation = await warehouseInstance.methods
+                .add_item(
+                    1,
+                    MichelsonMap.fromLiteral({
+                        XP: "97"
+                    }),
+                    9,
+                    "Karim Benzema",
+                    undefined,
+                    1
+                )
+                .send();
+
+            await operation.confirmation(1);
+        });
+
+        describe("And I assign it to an inventory", () => {
+            let oldInventoryStorage;
+            let newInventoryStorage;
+
+            beforeAll(async () => {
+                const operation = await warehouseInstance.methods
+                    .assign_item_proxy(oldInventoryInstance.address, 9, 1)
+                    .send();
+
+                await operation.confirmation(1);
+
+                oldInventoryStorage = await oldInventoryInstance.storage();
+                newInventoryStorage = await newInventoryInstance.storage();
+            });
+            
+            describe("When I transfer an item to an inventory that doesn't exist", () => {
+                it("Then fails with an explicit error", async () => {
```

Comment:
we should also add a test to ensure that the original contract still has the item